docs/development/codeguide.rst

```diff
@@ -134,7 +134,19 @@ Coding Style/Conventions
   using only 4 spaces for indentation, and never tabs.
 
 * Our testing infrastructure currently enforces a subset of the PEP8 style
-  guide. You can check locally whether your changes have followed these by
+  guide, and some packages enforce stronger styling checks such as using
+  `isort <https://pycqa.github.io/isort/>`_ to sort the module imports. We
+  provide a `pre-commit <https://pre-commit.com/>`_ hook that checks and enforces
+  all of these styling checks on each commit. We strongly suggest that you setup
+  and use the pre-commit hook, by running::
```

Comment:
Perhaps we should put a sentence here explaining what git pre-commit hooks actually are?

Reply:
Sure, I can do that. Should I include a link to the git docs about hooks?

Reply:
Doesn't hurt. Thanks!